crates/task-impls/src/network.rs

```diff
@@ -380,24 +380,35 @@ impl<
             HotShotEvent::QuorumVoteSend(vote) => {
                 *maybe_action = Some(HotShotAction::Vote);
                 let view_number = vote.view_number() + 1;
-                let leader = match self.quorum_membership.leader(view_number, self.epoch) {
-                    Ok(l) => l,
-                    Err(e) => {
-                        tracing::warn!(
-                            "Failed to calculate leader for view number {:?}. Error: {:?}",
-                            view_number,
-                            e
-                        );
-                        return None;
-                    }
+                let is_vote_leaf_extended = self
+                    .consensus
+                    .read()
+                    .await
+                    .is_leaf_extended(vote.data.leaf_commit);
+                let transmit_type = if is_vote_leaf_extended {
+                    tracing::debug!("We're voting for the last eQC proposal. Broadcast the vote.");
+                    TransmitType::Broadcast
+                } else {
+                    let leader = match self.quorum_membership.leader(view_number, self.epoch) {
+                        Ok(l) => l,
+                        Err(e) => {
+                            tracing::warn!(
+                                "Failed to calculate leader for view number {:?}. Error: {:?}",
+                                view_number,
+                                e
+                            );
+                            return None;
+                        }
+                    };
+                    TransmitType::Direct(leader)
```

Comment:
I feel like this logic is more appropriate in the quorum_vote task, maybe just add the ExtendedVoteSend and handle that event here instead